smoketests/packages/react-ts-esm/src/index.tsx

```diff
@@ -0,0 +1,69 @@
+import { BacktraceClient, ErrorBoundary } from '@backtrace/react';
+import { BacktraceSessionReplayModule } from '@backtrace/session-replay';
+import { useEffect, useState } from 'react';
+import ReactDOM from 'react-dom/client';
+
+const SUBMISSION_URL = new URLSearchParams(window.location.search).get('url');
+if (!SUBMISSION_URL) {
+    throw new Error('submission URL is required');
+}
+
+const client = BacktraceClient.initialize(
+    {
+        url: SUBMISSION_URL,
+        name: '@backtrace/react-example',
+        version: '0.0.1',
+        userAttributes: {
+            'custom-attribute': 'test',
+            'custom-annotation': {
+                prop1: true,
+                prop2: 123,
+            },
+        },
+    },
+    (builder) =>
+        builder.useModule(
+            new BacktraceSessionReplayModule({
+                maxEventCount: 100,
+            }),
+        ),
+);
+
+function App() {
+    const [status, setStatus] = useState('loading');
+    const [rxid, setRxid] = useState('');
+    const [error, setError] = useState('');
+
+    useEffect(() => {
```

Comment:
idea: if we can use buttons - we can click some of them and trigger actions instead of just veryfing what is the page content. Right now it's impossible to test against an error. Testign if the "message" is sent, is kinda useless.

Reply:
Added some new tests.